### PR TITLE
Nested attributes improvements

### PR DESCRIPTION
**Pull request by albertbellonch**
Hi @rogercampos,

I have implemented two new features for nested attributes:
- You can now use multiple nested associations, so you can have something like:

`save_for_dejavu @product, :nested => [:category, :colors]`
- You can now use any kind of association as nested: many-to-one, one-to-one, or one-to-many.

Of course, I added tests for everything.

### CODE REVIEW

File: test_app/app/controllers/products_controller.rb [hunk 1]
```diff
@@ -25,6 +25,8 @@ def show
   # GET /products/new.json
   def new
     @product = Product.new
+    @product.category = Category.new
```
Comment (rogercampos) [inline test_app/app/controllers/products_controller.rb:28]: Ala, this is 
    @product.build_category

Jeje

**Comment by rogercampos** [comment]
Nice! Thanks

**Comment by rogercampos** [comment]
I'll make a release when i get home

**Comment by albertbellonch** [comment]
Can you please make a release when you can?

Thanks!

**Comment by rogercampos** [comment]
Done! new 0.2.0 version released

**Comment by albertbellonch** [comment]
Very nice indeed!

**Pull request merged**